client/resource_group/controller/controller.go

```diff
@@ -429,19 +444,60 @@ func (c *ResourceGroupsController) sendTokenBucketRequests(ctx context.Context,
 // OnRequestWait is used to check whether resource group has enough tokens. It maybe needs to wait some time.
 func (c *ResourceGroupsController) OnRequestWait(
 	ctx context.Context, resourceGroupName string, info RequestInfo,
-) (*rmpb.Consumption, error) {
+) (*rmpb.Consumption, Delta, error) {
 	gc, err := c.tryGetResourceGroup(ctx, resourceGroupName)
 	if err != nil {
 		failedRequestCounter.WithLabelValues(resourceGroupName).Inc()
-		return nil, err
+		return nil, Delta{}, err
+	}
+	consumption, err := gc.onRequestWait(ctx, info)
+	if err != nil {
+		return nil, Delta{}, err
 	}
-	return gc.onRequestWait(ctx, info)
+
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	m, ok := c.resourceCounter[resourceGroupName]
+	if !ok {
+		m = Counter{
+			storeCounter:  make(map[uint64]Delta),
+			globalCounter: Delta{},
+		}
+		c.resourceCounter[resourceGroupName] = m
+	}
+	delta := Delta{
```

Comment:
What if StoreID is not existed before this request, I think in this case, it should be inited with the value of globalCounter, but here it is implicitly inited with 0. For new joined store, this delta value can be unexpectedly huge

Reply:
Oh, I miss it